libraries/NonFungibleToken/test/nonFungibleToken-tests.js

```diff
@@ -0,0 +1,368 @@
+
+			assert.equal(config.ownerKeyPair.publicKey, deployedContract.owner)
+		})
+	})
+
+	describe('Interact with contract', () => {
+		let deployedContract;
+		let compiledContract;
+
+		beforeEach(async () => {
+			compiledContract = await firstClient.contractCompile(contentOfContract, {
+				gas: config.gas
+			})
+			deployedContract = await compiledContract.deploy({
+				initState: `("${tokenName}", "${tokenSymbol}")`,
+				options: {
+					ttl: config.ttl
+				},
+				abi: "sophia"
+			});
+		})
+
+		describe('Read', () => {
+			it('call contract read successfully', async () => {
+				//Arrange
+
+				//Act
+				const callNamePromise = deployedContract.call('name', {
+					options: {
+						ttl: config.ttl
+					}
+				});
+				
+				const callNameResult = await callNamePromise;
+
+				const callSymbolPromise = deployedContract.call('symbol', {
+					options: {
+						ttl: config.ttl
+					}
+				});
+				
+				const callSymbolResult = await callSymbolPromise;
+
+				//Assert
+				const decodedNameResult = await callNameResult.decode("string");
+				const decodedSymbolResult = await callSymbolResult.decode("string");
+
+				assert.equal(decodedNameResult.value, tokenName)
+				assert.equal(decodedSymbolResult.value, tokenSymbol)
+			})
+		})
+
+		describe('Contract functionality', () => {
+			beforeEach(async () => {
+				const deployContractPromise = deployedContract.call('mint', {
+					args: `(${firstTokenId}, ${utils.publicKeyToHex(config.ownerKeyPair.publicKey)})`,
+					options: {
+						ttl: config.ttl
+					},
+					abi: "sophia"
+				})
+				
+				await deployContractPromise;
+			})
+
+			describe('Mint', () => {
+				it('should mint 1 token successfully', async () => {
+					//Arrange
+					const expectedBalance = 1;
+
+					//Act
+					const ownerOfPromise = deployedContract.call('ownerOf', {
+						args: `(${firstTokenId})`,
+						options: {
+							ttl: config.ttl
+						}
+					});
+					
+					const ownerOfResult = await ownerOfPromise;
+
+					const balanceOfPromise = deployedContract.call('balanceOf', {
+						args: `(${utils.publicKeyToHex(config.ownerKeyPair.publicKey)})`,
+						options: {
+							ttl: config.ttl
+						}
+					});
+					
+					const balanceOfResult = await balanceOfPromise;
+
+					//Assert
+					const decodedOwnerOfResult = await ownerOfResult
+					let encodedData = await decodedOwnerOfResult.decode('address')
+					const ownerPublicKey = crypto.aeEncodeKey(bytes.toBytes(encodedData.value, true))
+
+					const decodedBalanceOfResult = await balanceOfResult.decode("int");
+
+					assert.equal(ownerPublicKey, config.ownerKeyPair.publicKey)
+					assert.equal(decodedBalanceOfResult.value, expectedBalance)
+				})
+
+				it('should not mint from non-owner', async () => {
+					const unauthorisedPromise = secondClient.contractCall(compiledContract.bytecode, 'sophia', deployedContract.address, "mint", {
+						args: `(${firstTokenId}, ${utils.publicKeyToHex(config.ownerKeyPair.publicKey)})`,
```

Comment:
I see a lot of repetition of the value inside args (sometimes without firstTokenId), maybe is a good idea to take this to a variable in the before?